auth/src/main/java/com/firebase/ui/auth/data/remote/TwitterSignInHandler.java

```diff
@@ -0,0 +1,119 @@
+package com.firebase.ui.auth.data.remote;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.model.User;
+import com.firebase.ui.auth.viewmodel.idp.ProviderHandlerBase;
+import com.firebase.ui.auth.viewmodel.idp.ProviderHandlerParamsBase;
+import com.firebase.ui.auth.viewmodel.idp.ProvidersHandlerBase;
+import com.google.firebase.auth.TwitterAuthProvider;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.Twitter;
+import com.twitter.sdk.android.core.TwitterAuthConfig;
+import com.twitter.sdk.android.core.TwitterConfig;
+import com.twitter.sdk.android.core.TwitterCore;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.TwitterSession;
+import com.twitter.sdk.android.core.identity.TwitterAuthClient;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class TwitterSignInHandler extends ProviderHandlerBase<TwitterSignInHandler.Params> {
+    private final TwitterAuthClient mClient;
+    private final Callback mCallback = new Callback();
+
+    public TwitterSignInHandler(Application application) {
+        super(application);
+        initialize();
+        mClient = new TwitterAuthClient();
+    }
+
+    public static void signOut() {
+        try {
+            TwitterCore.getInstance();
+        } catch (IllegalStateException e) {
+            initialize();
+        }
+
+        TwitterCore.getInstance().getSessionManager().clearActiveSession();
+    }
+
+    private static void initialize() {
+        Context context = AuthUI.getApplicationContext();
```

Comment:
Could this be an instance method instead of a static method and use the `Context` passed into the constructor instead of the `AuthUI` application context?  I realize this would affect `signOut` as well but do we sign out from anywhere that we couldn't create a new instance?

Reply:
My main motivation for asking is the strangeness of a `static initialize()` method.